cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/batch/mapreduce/MapReduceRuntimeContext.java

```diff
@@ -98,4 +98,36 @@ public T call() throws DatasetInstantiationException {
       throw Throwables.propagate(e);
     }
   }
+
+  @Override
+  public void releaseDataset(final Dataset dataset) {
+    try {
+      LogContext.runWithoutLogging(new Callable<Void>() {
+        @Override
+        public Void call() {
+          context.releaseDataset(dataset);
+          return null;
+        }
+      });
+    } catch (Exception e) {
+      Throwables.propagateIfInstanceOf(e, DatasetInstantiationException.class);
+      throw Throwables.propagate(e);
+    }
+  }
+
+  @Override
+  public void discardDataset(final Dataset dataset) {
+    try {
+      LogContext.runWithoutLogging(new Callable<Void>() {
+        @Override
+        public Void call() {
+          context.discardDataset(dataset);
+          return null;
+        }
+      });
+    } catch (Exception e) {
+      Throwables.propagateIfInstanceOf(e, DatasetInstantiationException.class);
```

Comment:
This is not needed since `DatasetInstantiationException` is a runtime exception, the `throw Throwables.propagte(e);` will throw it as is without wrapping.